### PR TITLE
[postgres] close jdbc connection when close replication connection

### PR DESCRIPTION
**Pull request by loserwang1024**
### What is the purpose of the change
In PostgresSourceEnumerator#createSlotForGlobalStreamSplit, PostgresReplicationConnection will be closed after create slot.
However, not any PostgresReplicationConnection.close will close the PostgresConnection in messageDecoder, which is generated by Hikari pool and is passed to PostgresReplicationConnection.

See AbstractMessageDecoder.close:
*  NonStreamingWal2JsonMessageDecoder.close： do nothing
* PgProtoMessageDecoder.close: do nothing
* StreamingWal2JsonMessageDecoder.close: do nothing
* PgOutputMessageDecoder.close: **the only one to close jdbc connection**.

Thus, the Hikari pool will be full if there is too much pg source.

### Verified this problem
Debug  PostgresReplicationConnection.close in any test.

### CODE REVIEW

**Comment by loserwang1024** [comment]
@TyrantLucifer CC

**Review by ruanhang1993 — APPROVED** [review]
LGTM.

File: flink-connector-postgres-cdc/src/main/java/com/ververica/cdc/connectors/postgres/source/enumerator/PostgresSourceEnumerator.java [hunk 1]
```diff
@@ -78,9 +78,9 @@ private void createSlotForGlobalStreamSplit() {
             return;
         }
 
-        try {
+        try (PostgresConnection connection = postgresDialect.openJdbcConnection()) {
```
Comment (ruanhang1993) [inline flink-connector-postgres-cdc/src/main/java/com/ververica/cdc/connectors/postgres/source/enumerator/PostgresSourceEnumerator.java:81]: Maybe we could reuse the same connection in line 64.

**Review by ruanhang1993 — APPROVED** [review]
LGTM.

**Review by TyrantLucifer — COMMENTED** [review]
LGTM

**Pull request merged**